### PR TITLE
add layla

### PR DESCRIPTION
**Pull request by imring**
- [x] Frames
- [x] Hitlags
- [x] Talents
- [x] C0 Normals
- [x] C0 Skill
- [x] C0 Burst
- [x] A1
- [x] A4
- [x] C1
- [x] C2
- [x] C3
- [x] C4
- [x] C5
- [x] C6

### CODE REVIEW

**Comment by RarePossum** [comment]
Don't forget to add her to HP scaling in substat optimiser

File: internal/characters/layla/shield.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+			sort.Slice(trgs, func(i, j int) bool { return i < j })
+			nearTarget = trgs[0]
+		}
+
+		if last == -1 {
+			// if not found
+			if nearTarget == -1 {
+				c.Core.Tasks.Add(c.ShootStars(src, -1), 0.1*60)
+				return
+			} else {
```
Comment (k0l11) [inline internal/characters/layla/shield.go:109]: Please reduce nesting by putting the stuff inside of the else right below the if statement and then removing the else.

File: internal/characters/layla/shield.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+			if c.Base.Cons >= 2 {
+				c.AddEnergy("layla-c2", 1)
+			}
+		}
+
+		c.Core.QueueAttack(ai, combat.NewCircleHit(c.Core.Combat.Enemy(nearTarget), 0.5), 0, starsTravel[len(starsTravel)-stars], cb)
+
+		stars--
+		c.SetTag(nightStars, stars)
+		if stars == 0 {
```
Comment (k0l11) [inline internal/characters/layla/shield.go:162]: Please reduce nesting by adding an early return here and by putting the stuff inside of the else outside and then removing the else.

File: internal/characters/layla/shield.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+			c.RemoveTag(shootingStars)
+			c.starTickSrc = c.Core.F
+			c.TickNightStar(c.starTickSrc, false)
+		} else {
+			c.Core.Tasks.Add(c.ShootStars(src, nearTarget), 0.45*60)
+		}
+	}
+}
+
+func (c *char) TickNightStar(src int, star bool) func() {
```
Comment (k0l11) [inline internal/characters/layla/shield.go:172]: Capital letter in the beginning changes visibility, so please rename it to tickNightStar.

File: internal/characters/layla/skill.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(skillFrames),
+		AnimationLength: skillFrames[action.InvalidAction],
+		CanQueueAfter:   skillFrames[action.ActionDash], // earliest cancel is before skillHitmark
+		State:           action.SkillState,
+	}
+}
+
+func (c *char) StarsSkill() {
```
Comment (k0l11) [inline internal/characters/layla/skill.go:78]: Capital letter in the beginning changes visibility, so please rename it to starsSkill.

File: internal/characters/layla/burst.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		AttackTag:  combat.AttackTagElementalBurst,
+		ICDTag:     combat.ICDTagElementalBurst,
+		ICDGroup:   combat.ICDGroupDefault,
+		StrikeType: combat.StrikeTypeDefault,
+		Element:    attributes.Cryo,
+		Durability: 25,
+		FlatDmg:    burst[c.TalentLvlBurst()] * c.MaxHP(),
+	}
+
+	c.Core.Status.Add("laylaburst", 12*60+burstStart)
```
Comment (k0l11) [inline internal/characters/layla/burst.go:49]: NIT: Please consider naming status in all files to match "layla-e-..." and "layla-q-..." or at least be consistent on the hyphens ("layla-c4" vs. "laylaburst").

File: internal/characters/layla/skill.go [hunk 6]
```diff
@@ -41,45 +40,46 @@ func (c *char) Skill(p map[string]int) action.ActionInfo {
...
 		if !ok {
 			panic("target 0 should be Player but is not!!")
 		}
 		player.ApplySelfInfusion(attributes.Cryo, 25, 0.1*60)
 
 		c.starTickSrc = c.Core.F
-		c.TickNightStar(c.starTickSrc, false)
-	}, 32)
+		c.tickNightStar(c.starTickSrc, false)
+	}, 6)
```
Comment (k0l11) [inline internal/characters/layla/skill.go:66]: this should be 19 instead of 6

**Review by k0l11 — APPROVED** [review]

**Pull request merged**